lua/lsp/null-ls/services.lua

```diff
@@ -1,29 +1,23 @@
 local M = {}
 
-local logger = require("core.log"):get_default()
-
 local function find_root_dir()
-  if lvim.builtin.rooter.active then
-    --- use vim-rooter to set root_dir
-    vim.cmd "let root_dir = FindRootDirectory()"
-    return vim.api.nvim_get_var "root_dir"
-  end
-
-  -- TODO: Rework this to not make it javascript specific
-  --- use LSP to set root_dir
+  local util = require "lspconfig/util"
   local lsp_utils = require "lsp.utils"
-  local ts_client = lsp_utils.get_active_client_by_ft "typescript"
```

Comment:
Maybe we need a new utility function to check if any of the active language servers' root directory contain `node_modules`. Something like this...
```lua
function M.get_lsp_node_modules_dir()
  local clients = vim.lsp.get_active_clients()
  for _, client in pairs(clients) do
    local node_modules = client.config.root_dir .. "/node_modules"
    if vim.fn.glob(node_modules) ~= "" then
      return node_modules
    end
  end
  return nil
end
```

Reply:
> I do not think tsserver is always active when e.g. local prettier is needed. 

I don't need it to be active, that's why the `root_pattern`call is there. I'm doing a look up for tsserver just once, since it's the common one. Otherwise we're completely independent from any language server. I tested this code with markdown and it works correctly with a local `prettier`.

Reply:
> Maybe we need a new utility function to check if any of the active language servers' root directory contain `node_modules`. Something like this...
> ```lua
> function M.get_lsp_node_modules_dir()
>   local clients = vim.lsp.get_active_clients()
>   for _, client in pairs(clients) do
>     local node_modules = client.config.root_dir .. "/node_modules"
>     if vim.fn.glob(node_modules) ~= "" then
>       return node_modules
>     end
>   end
>   return nil
> end
> ```

I'm pretty confident that libuv is plenty fast and there's no need for this lookup. What happens if you end up in a different node_modules folder? I know it's possible with tsserver alone, but it would be easy to spot if it becomes problematic. 

My point is, treat the tsserver check as a cached value and no more, we can even remove it completely and it would still be fine :)